src/Sulu/Bundle/AdminBundle/Resources/js/containers/FieldBlocks/types.js

```diff
@@ -2,6 +2,11 @@
 import type {Node} from 'react';
 import type {SchemaEntry} from '../Form/types';
 
+export type BlockEntry<T> = {
```

Comment:
this is the type for the `FieldBlocks`, right? is there a reason for making this generic?

Reply:
Yeah, the `FieldBlocks` are used for the schedules in the overlay, and this way we can say that only `fixed` and `weekly` is allowed as type, not any string. Sadly this does not work all the way, but in some parts it is already useful.

Reply:
maybe i missed something, but i think the `Schedules` does use only the `BlockCollection`, not the `FieldBlocks` 🤔

Reply:
Right, it seems like it indeed would not be absolutely necessary in this PR. IIRC it was necessary in the beginning of my PR, but some refactoring seems to make it obsolete.

But don't you think it would be a good idea anyway? Would allow for better typing when the `FieldBlocks` component is used in some other place.

Reply:
i dont think it changes anything when the `FieldBlocks` is used in some other place. The `FieldBlock` sets the generic parameter anyway (`extends React.Component<FieldTypeProps<Array<BlockEntry<string>>>>`). Would make this non-generic because it is easier to understand 🙂